File: answers/challenge_2.java
package com.bbo.examtime;

public class challenge_2 {
    public int addDigits(int num) {
        while(num / 10 > 0) {
            int a = 0;
            while(num > 0) {
                a += num % 10;
                num /= 10;
            }
            num = a;
        }
        return num;
    }
}

File: Answers/challenge_2.php
<?php

class Solution
{

    /**
     * @param $num
     * @return float|int|mixed
     */
    function addDigits($num)
    {
        return $num == 0 ? $num : $num - (9) * ((int)floor(($num - 1) / 9));
    }
}